sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/DefaultDocumentQueryExecutionContext.java

```diff
@@ -186,7 +186,13 @@ protected PartitionKeyInternal getPartitionKeyInternal() {
                                                     this.fetchExecutionRangeAccumulator.getExecutionRanges(),
                                                     Arrays.asList(schedulingTimeSpanMap)),
                                             tFeedResponse.getActivityId());
-                            BridgeInternal.putQueryMetricsIntoMap(tFeedResponse, DEFAULT_PARTITION_RANGE, qm);
+                            String pkrId = tFeedResponse.getResponseHeaders().get(HttpConstants.HttpHeaders.PARTITION_KEY_RANGE_ID);
+                            if (StringUtils.isEmpty(pkrId)) {
+                                pkrId = "0";
```

Comment:
on which scenario we come here? is this code reachable?

can't we rely on the targetRage see my above comment for this?

Reply:
Removed this if condition